File: 1353-find-resultant-array-after-removing-anagrams/1353-find-resultant-array-after-removing-anagrams.py
class Solution:
    def removeAnagrams(self, words):
        result = []
        prev_sorted = ""

        for w in words:
            sorted_word = ''.join(sorted(w))
            if sorted_word != prev_sorted:
                result.append(w)
                prev_sorted = sorted_word
        return result
    

File: 1353-find-resultant-array-after-removing-anagrams/1353-find-resultant-array-after-removing-anagrams.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        vector<string> result;
        string prevSorted = "";

        for (string word : words) {
            string sorted = word;
            sort(sorted.begin(), sorted.end());

            if (sorted != prevSorted) {
                result.push_back(word);
                prevSorted = sorted;
            }
        }
        return result;
    }
};
